server/src/main/java/org/elasticsearch/cluster/metadata/OperatorMetadata.java

```diff
@@ -0,0 +1,292 @@
+     *
+     * @param in the {@link StreamInput} to read from
+     * @return {@link OperatorMetadata}
+     * @throws IOException
+     */
+    public static OperatorMetadata readFrom(StreamInput in) throws IOException {
+        Builder builder = new Builder(in.readString()).version(in.readLong());
+
+        int handlersSize = in.readVInt();
+        for (int i = 0; i < handlersSize; i++) {
+            OperatorHandlerMetadata handler = OperatorHandlerMetadata.readFrom(in);
+            builder.putHandler(handler);
+        }
+
+        builder.errorMetadata(in.readOptionalWriteable(OperatorErrorMetadata::readFrom));
+        return builder.build();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(namespace);
+        out.writeLong(version);
+        out.writeCollection(handlers.values());
+        out.writeOptionalWriteable(errorMetadata);
+    }
+
+    /**
+     * Reads an {@link OperatorMetadata} {@link Diff} from {@link StreamInput}
+     *
+     * @param in the {@link StreamInput} to read the diff from
+     * @return a {@link Diff} of {@link OperatorMetadata}
+     * @throws IOException
+     */
+    public static Diff<OperatorMetadata> readDiffFrom(StreamInput in) throws IOException {
+        return SimpleDiffable.readDiffFrom(OperatorMetadata::readFrom, in);
+    }
+
+    /**
+     * Empty {@link org.elasticsearch.cluster.DiffableUtils.MapDiff} helper for metadata backwards compatibility.
+     */
+    public static final DiffableUtils.MapDiff<String, OperatorMetadata, Map<String, OperatorMetadata>> EMPTY_DIFF =
+        new DiffableUtils.MapDiff<>(null, null, List.of(), List.of(), List.of()) {
+            @Override
+            public Map<String, OperatorMetadata> apply(Map<String, OperatorMetadata> part) {
+                return part;
+            }
+        };
+
+    /**
+     * Convenience method for creating a {@link Builder} for {@link OperatorMetadata}
+     *
+     * @param namespace the namespace under which we'll store the {@link OperatorMetadata}
+     * @return {@link Builder}
+     */
+    public static Builder builder(String namespace) {
+        return new Builder(namespace);
+    }
+
+    /**
+     * Convenience method for creating a {@link Builder} for {@link OperatorMetadata}
+     *
+     * @param namespace the namespace under which we'll store the {@link OperatorMetadata}
+     * @param metadata an existing {@link OperatorMetadata}
+     * @return {@link Builder}
+     */
+    public static Builder builder(String namespace, OperatorMetadata metadata) {
+        return new Builder(namespace, metadata);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(namespace());
+        builder.field(VERSION.getPreferredName(), version);
+        builder.startObject(HANDLERS.getPreferredName());
+        var sortedKeys = new TreeSet<>(handlers.keySet());
```

Comment:
Instead of using a TreeSet here to do the sorting, we can use the stream api?

```
handlers.entrySet().stream().sorted(Map.Entry.comparingByKey()).forEach(e -> {
    e.getValue().toXContent(builder, params);
}
```